src/cbc_sdk/enterprise_edr/threat_intelligence.py

```diff
@@ -254,17 +417,57 @@ def reports(self):
 
         return reports_
 
+    def add_report_ids(self, report_ids):
+        """
+        Adds new report IDs to the watchlist.
+
+        Args:
+            report_ids (list[str]): List of report IDs to be added to the watchlist.
+        """
+        old_report_ids = self.report_ids if self._info.get('report_ids') else []
+        self.update(report_ids=(old_report_ids + report_ids))
+
+    def add_reports(self, reports):
+        """
+        Adds new reports to the watchlist.
+
+        Args:
+            reports (list[Report]): List of reports to be added to the watchlist.
+        """
+        report_ids = []
+        for report in reports:
+            report.validate()
+            if report._from_watchlist:
+                report_ids.append(report._info['id'])
+        self.add_report_ids(report_ids)
+
 
 class Feed(FeedModel):
     """Represents an Enterprise EDR feed's metadata."""
     urlobject = "/threathunter/feedmgr/v2/orgs/{}/feeds"
     urlobject_single = "/threathunter/feedmgr/v2/orgs/{}/feeds/{}"
     primary_key = "id"
     swagger_meta_file = "enterprise_edr/models/feed.yaml"
-
-    @classmethod
-    def _query_implementation(self, cb, **kwargs):
-        return FeedQuery(self, cb)
+    REPORT_VALIDATION = [
```

Comment:
I don't know what "schemas" are. I just did what was necessary to try and keep the validation functions from getting out of hand.

Reply:
I provided a link which creates the schemas we are using to do the same validation in our STIX/TAXII example.  Please review and give me your thoughts as the custom function and design will be harder to maintain than something we use else where in the repo

Reply:
I see that file, but all it does is create Schema objects, it doesn't show how they are used. Could you provide a link to the documentation for the schema package so I at least have some clue how those objects work?

Reply:
That information was just what I needed.

The use of `schema` shrinks down the code a lot, but, oh, Lord, `schema` generates _UGLY_ error messages by default. I wound up having to override virtually every error message that could be thrown by the schema and tweak the way I get the message out of the `SchemaError` for the `InvalidObjectError`.  This is in keeping with your desire for "descriptive and intuitive" error handling.  It also makes the unit test function _way_ more complicated, or, rather, its `parametrize` blocks, but, after much effort, I got everything synced back up again.  I also had to add `schema` to the `requirements.txt` file; I hope that won't be a problem.

Reply:
Thanks! This will be a lot easier to maintain and expand in the future. The package add makes sense as we will probably use it in more places in the future